cql3/functions/functions.cc

```diff
@@ -126,17 +122,19 @@ void functions::add_function(shared_ptr<function> func) {
 
 template <typename F>
 void functions::with_udf_iter(const function_name& name, const std::vector<data_type>& arg_types, F&& f) {
-    auto i = find_iter(name, arg_types);
-    if (i == _declared.end() || i->second->is_native()) {
+    auto cit = find_iter(name, arg_types);
+    if (cit == _declared.end() || cit->second->is_native()) {
         log.error("attempted to remove or alter non existent user defined function {}({})", name, arg_types);
         return;
     }
-    f(i);
+    // erase here is only to convert from const_iterator to iterator
+    auto it = _declared.erase(cit, cit);
+    f(it);
```

Comment:
I'd introduce non-const `find_iter` or consider `const_cast` (which has some problems of its own)

Reply:
> I'd introduce non-const find_iter or consider const_cast (which has some problems of its own)

You're proposing some alternative solutions but not providing any arguments behind it.

I think it's reasonable compromise, to use erase with appropriate comment. If you search the internet for this problem that's common suggestion.

Reply:
Apparently I searched the Internet the other way round, because it gives me different results ;) My reasoning is based on top of https://isocpp.org/wiki/faq/const-correctness#const-overloading, so introducing non const counterpart function. As for the way I comment, I just wanted to voice my opinion, because I share the same concerns as @piodul, but I won't insist on anything, I don't think this is super important.